addons/mail/static/src/js/activity.js

```diff
@@ -349,6 +350,35 @@ var Activity = AbstractActivityField.extend({
             }).popover('show');
         }
     },
+    _onClickMailTemplate: function (event) {
+        var templateId = $(event.target).data('template-id');
+        var activityID = $(event.target).parent().data('activity-id');
+        var previousActivityTypeID = $(event.target).parent().data('previous-activity-type-id');
+        var self = this;
+        var action = {
+            name: _('Compose Email'),
+            type: 'ir.actions.act_window',
+            res_model: 'mail.compose.message',
+            views: [[false, 'form']],
+            target: 'new',
+            context: {
+                default_res_id: this.res_id,
+                default_model: this.model,
+                default_use_template: true,
+                default_template_id: templateId,
+                force_email: true,
+                from_activity: true,
+            },
+        };
+        return this.do_action(action, { on_close: function(infos) {
```

Comment:
Generic for this JS addition: shouldn't we add a test :postal_horn:  ? At least checking that clicking on a template name correctly triggers the action. Something cool like that. Not sure how to check it and catch the action of the do_action though.